pkg/controller/clusterpoolnamespace/clusterpoolnamespace_controller.go

```diff
@@ -0,0 +1,157 @@
+package clusterpoolnamespace
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/openshift/hive/pkg/constants"
+	hivemetrics "github.com/openshift/hive/pkg/controller/metrics"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+)
+
+const (
+	ControllerName                                 = "clusterpoolnamespace"
+	minimumLifetime                                = 5 * time.Minute
+	durationBetweenDeletingClusterDeploymentChecks = 1 * time.Minute
+)
+
+// Add creates a new ClusterDeployment Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return AddToManager(mgr, NewReconciler(mgr))
+}
+
+// NewReconciler returns a new reconcile.Reconciler
+func NewReconciler(mgr manager.Manager) reconcile.Reconciler {
+	r := &ReconcileClusterPoolNamespace{
+		Client: controllerutils.NewClientWithMetricsOrDie(mgr, ControllerName),
+		logger: log.WithField("controller", ControllerName),
+	}
+	return r
+}
+
+// AddToManager adds a new Controller to mgr with r as the reconcile.Reconciler
+func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(
+		fmt.Sprintf("%s-controller", ControllerName),
+		mgr,
+		controller.Options{
+			Reconciler:              r,
+			MaxConcurrentReconciles: controllerutils.GetConcurrentReconciles(),
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to Namespaces
+	if err := c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{}); err != nil {
```

Comment:
Should we limit to just those with our label?

Reply:
We **can**. It is quite a bit more code to do it in a predicate rather than doing it in `Reconcile`. Do we have a strong reason for wanting to use a predicate?